bugfix to allow optionals in BudgetSummary model

TheFern2 opened the pull request:
This PR is in regards to #12 It makes those two properties as Optionals when they are null from the api json response.

[comment describing issue](https://github.com/andrebocchini/swiftynab/issues/12#issuecomment-2471886389)

I can now get budgets with no error:
```
Content view launched
Token ZOgP_76Ov5pv7fbbw6zA8U-sSxJYzwo4r_h79COZ4UU
Logged in status: true
2024_Round2 (Archived on 2024-08-17)
My Budget (Archived on 2022-03-20)
New2 Budget
My Budget
2024_Round3 (Archived on 2024-09-18)
2024TakingCharge:( (Archived on 2024-02-02)
2024_Round4
My Budget
My Budget
2021-2023 Transactions
New1 Budget (Archived on 2022-04-27) (Archived on 2023-07-05)
2023TakingCharge (Archived on 2024-01-04)
```

andrebocchini commented:
I can merge this as is so you get credit for the contribution. There should still be some version updates to other files, but I'll do those and then publish a release.

TheFern2 commented:
Hey thanks for merging.I'm still working on my iOS app I've been away for a while with health issues but I'll make another pr at some point.

I've added a bunch of improvements mainly for mock up initialization and then I've fixed functions to actually return server knowledge which is something that bug me for delta requests.

As I continue working on my app I'll keep updating this lib on init methods mainly.

TheFern2 commented:
Here's my latest working branch https://github.com/TheFern2/swiftynab/tree/delta-responses

Btw I found that if you change a budget group category name ynab api changes server knowledge without returning new data they say is not a bug but it definitely is. I submitted a feature request but I would probably add a force sync button to my app.
